feat: set default icon color to #0E353D

codeheroics opened the pull request:
This proposal aims to change the default color of icons to #0E353D.

The current behaviour is that icons have a default values for the `color`, `activeColor`, and `hoverColor` props of `inherit`, which is supposed to take the value of a parent `fill` value and apply it. 

In reality, as `fill` is a SVG property that will not be defined on any parent container, this property is useless and equivalent to having the default color set to the browser default, #000.

This PR would thus set icons to the default dashlane dark green when no color is specified.

(I am unable to build the docs for this, `yarn docz:build` removes files but do not create any, I have the same issue on master)

maryiam commented:
Are you sure that this change suits for all projects? This is something to review with designers because I'm not sure if DashGreen should always the default value.

maryiam commented:
> (I am unable to build the docs for this, `yarn docz:build` removes files but do not create any, I have the same issue on master)

What kind of error do you have when you build?

maryiam commented on src/ions/icon/base/Icon.tsx:
```diff
@@ -9,7 +10,7 @@ interface Props extends Omit<IconProps, 'title' | 'children'> {
 const getHeight = ({ height, size }: Props) => `${size || height || 20}px`;
 const getColor = (propName: ColorProps = 'color') => (props: Props): string =>
-  props.animated ? 'none' : props[propName] || props['color'] || 'inherit';
+  props.animated ? 'none' : props[propName] || props['color'] || dashGreen00;
```
Pay attention here because `dashGreen00 ` will be set as default color for hover and active states if no props are passed.

codeheroics replied:
I'm not sure I get your point, this is the intended behaviour. Currently default icons are black and stay black on hover / active, the idea here is to change the default for all 3 states, normal, hover & active

codeheroics commented:
> What kind of error do you have when you build?

No error, but it deletes a lot of files and doesn't create any new ones (and I tried reinstalling my yarn dependencies, but nothing changed)

```
        modified:   docs/assets.json
        modified:   docs/index.html
        deleted:    docs/static/css/src-components-alert-alert.4037b66c5faa136c379c.css
        deleted:    docs/static/css/src-components-text-input-text-input.4037b66c5faa136c379c.css
        deleted:    docs/static/js/app.4037b66c5faa136c379c.js
        deleted:    docs/static/js/app.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/documentation-color-stepbystep.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/documentation-color-stepbystep.ee8cfa7f.js
        deleted:    docs/static/js/documentation-component-stepbystepguide.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/documentation-component-stepbystepguide.b45e7fdb.js
        deleted:    docs/static/js/documentation-icon-stepbystep.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/documentation-icon-stepbystep.997df3fe.js
        deleted:    docs/static/js/src-atoms-button-button.08fe320a.js
        deleted:    docs/static/js/src-atoms-button-button.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/src-atoms-icons-icons.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/src-atoms-icons-icons.5504c5d3.js
        deleted:    docs/static/js/src-components-alert-alert.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/src-components-alert-alert.ad07c9d7.js
        deleted:    docs/static/js/src-components-dropdown-dropdown-menu.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/src-components-dropdown-dropdown-menu.d1cab00e.js
        deleted:    docs/static/js/src-components-dropdown-dropdown-menu~src-components-quick-actions-menu-quick-actions-menu.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/src-components-dropdown-dropdown-menu~src-components-quick-actions-menu-quick-actions-menu.d72e4123.js
        deleted:    docs/static/js/src-components-logo-logo.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/src-components-logo-logo.591e6af0.js
        deleted:    docs/static/js/src-components-quick-actions-menu-quick-actions-menu.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/src-components-quick-actions-menu-quick-actions-menu.9a62d83d.js
        deleted:    docs/static/js/src-components-text-input-text-input.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/src-components-text-input-text-input.a6168bb0.js
        deleted:    docs/static/js/src-components-tooltip-tooltip.3d9fcf35.js
        deleted:    docs/static/js/src-components-tooltip-tooltip.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/src-index.4037b66c5faa136c379c.js.map
        deleted:    docs/static/js/src-index.dc156c75.js
        deleted:    docs/static/js/vendors.31701a81.js
        deleted:    docs/static/js/vendors.4037b66c5faa136c379c.js.map
```

codeheroics commented:
> Are you sure that this change suits for all projects? This is something to review with designers because I'm not sure if DashGreen should always the default value.

Currently trying to get an answer about this :slightly_smiling_face:

maryiam commented:
> > What kind of error do you have when you build?
> 
> No error, but it deletes a lot of files and doesn't create any new ones (and I tried reinstalling my yarn dependencies, but nothing changed)
> 
> ```
>         modified:   docs/assets.json
>         modified:   docs/index.html
>         deleted:    docs/static/css/src-components-alert-alert.4037b66c5faa136c379c.css
>         deleted:    docs/static/css/src-components-text-input-text-input.4037b66c5faa136c379c.css
>         deleted:    docs/static/js/app.4037b66c5faa136c379c.js
>         deleted:    docs/static/js/app.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/documentation-color-stepbystep.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/documentation-color-stepbystep.ee8cfa7f.js
>         deleted:    docs/static/js/documentation-component-stepbystepguide.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/documentation-component-stepbystepguide.b45e7fdb.js
>         deleted:    docs/static/js/documentation-icon-stepbystep.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/documentation-icon-stepbystep.997df3fe.js
>         deleted:    docs/static/js/src-atoms-button-button.08fe320a.js
>         deleted:    docs/static/js/src-atoms-button-button.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/src-atoms-icons-icons.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/src-atoms-icons-icons.5504c5d3.js
>         deleted:    docs/static/js/src-components-alert-alert.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/src-components-alert-alert.ad07c9d7.js
>         deleted:    docs/static/js/src-components-dropdown-dropdown-menu.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/src-components-dropdown-dropdown-menu.d1cab00e.js
>         deleted:    docs/static/js/src-components-dropdown-dropdown-menu~src-components-quick-actions-menu-quick-actions-menu.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/src-components-dropdown-dropdown-menu~src-components-quick-actions-menu-quick-actions-menu.d72e4123.js
>         deleted:    docs/static/js/src-components-logo-logo.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/src-components-logo-logo.591e6af0.js
>         deleted:    docs/static/js/src-components-quick-actions-menu-quick-actions-menu.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/src-components-quick-actions-menu-quick-actions-menu.9a62d83d.js
>         deleted:    docs/static/js/src-components-text-input-text-input.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/src-components-text-input-text-input.a6168bb0.js
>         deleted:    docs/static/js/src-components-tooltip-tooltip.3d9fcf35.js
>         deleted:    docs/static/js/src-components-tooltip-tooltip.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/src-index.4037b66c5faa136c379c.js.map
>         deleted:    docs/static/js/src-index.dc156c75.js
>         deleted:    docs/static/js/vendors.31701a81.js
>         deleted:    docs/static/js/vendors.4037b66c5faa136c379c.js.map
> ```

Can you please try to completely delete the `docs` folder, rebuild the doc with `yarn docz:build` and commit it?

codeheroics commented:
@maryiam ok I figured it out! The `docs` folder is ignored in .gitignore (is it normal?) so I had to temporarily disable it to be able to commit the new files.

maryiam commented:
> @maryiam ok I figured it out! The `docs` folder is ignored in .gitignore (is it normal?) so I had to temporarily disable it to be able to commit the new files.

Yes this is normal. It's explained in the Contributing file:
`We also execute the task yarn exec-docz-update on post-commit that will check if we have added or updated any Docz documentation (by modifying or adding any .mdx file), regenerate new built documentation and add push it through a new dedicated commit : in this way, we assure that https://dashlane.github.io/ui-components is always up to date.`

So you are not supported to do this manually, the update docs commit is triggered everytime you modify something kind `MDX` files.

pducolin commented on src/ions/icon/types.ts:
```diff
@@ -34,17 +34,17 @@ export interface Props {
   /**
    * The icon color
-   * @default 'inherit' (parent's fill property)
+   * @default '#0E353D' (dashlane default dark green)
```
I would use only the css var name here

oclero replied:
Yes, please no hardcoded color like this, or the goal of the Design System is destroyed haha

codeheroics replied:
Done, the comment now says `@default dashGreen00 (dashlane default dark green)` :slightly_smiling_face:

codeheroics commented:
> > Are you sure that this change suits for all projects? This is something to review with designers because I'm not sure if DashGreen should always the default value.
> 
> Currently trying to get an answer about this slightly_smiling_face

Got designers approval :slightly_smiling_face:
